code/parse/scripting.cpp

```diff
@@ -427,6 +428,11 @@ bool ConditionedHook::ConditionsValid(int action, object *objp, int more_data)
 									return false;
 								break;
 							}
+							case CHA_BEAMFIRE: {
+								if (!(stricmp(Weapon_info[more_data].name, scp->data.name) == 0))
```

Comment:
Well, there's a lot of "methods" used in the code base for stricmp and I think I copy pasted the method used in CHA_ONTURRETFIRED a few lines up. It's a quick change anyhow...